pkg/router/httpTriggers.go

```diff
@@ -395,5 +406,10 @@ func (ts *HTTPTriggerSet) updateRouter() {
 
 		// make a new router and use it
 		ts.mutableRouter.updateRouter(ts.getRouter(functionTimeout))
+
+		if !ts.hasFunctionsAndTriggers.Load() && len(ts.triggers) > 0 && len(ts.functions) > 0 {
+			// let the startup probe know that the router is ready
+			ts.hasFunctionsAndTriggers.Store(true)
+		}
```

Comment:
The reason I check for `len(ts.triggers) > 0 && len(ts.functions) > 0` is because the first iteration of this loop doesn't have any functions or triggers.

I added a `updating router` debug log while testing to (eventually) figure this out...

```
2023-08-11T15:28:27.190Z	DEBUG	utils/namespace.go:49	namespaces	{"function_namespace": "", "builder_namespace": "", "default_namespace": "", "fission_resource_namespace": ["default", "gadget-app-sandbox-development", "gadget-app-sandbox-test"]}
2023-08-11T15:28:27.624Z	INFO	crd/client.go:111	Waiting for CRDs to be installed
2023-08-11T15:28:27.684Z	INFO	router/router.go:214	starting router	{"port": 8888}
2023-08-11T15:28:27.688Z	INFO	httpserver/server.go:21	starting server	{"service": "metrics", "addr": ":8080"}
2023-08-11T15:28:27.694Z	INFO	httpserver/server.go:21	starting server	{"service": "router", "addr": ":8888"}
2023-08-11T15:28:47.716Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:412	updating router	{"triggers": 0, "functions": 0}
2023-08-11T15:28:47.723Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:412	updating router	{"triggers": 3, "functions": 3}
2023-08-11T15:28:47.726Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:214	add prefix route for function	
2023-08-11T15:28:47.756Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:234	add handler route for function	
2023-08-11T15:28:47.756Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:214	add prefix route for function	
2023-08-11T15:28:47.757Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:273	add internal handler and prefix route for function
2023-08-11T15:28:47.757Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:273	add internal handler and prefix route for function
2023-08-11T15:28:47.757Z	DEBUG	triggerset.http_trigger_set	router/httpTriggers.go:273	add internal handler and prefix route for function
```

Reply:
Well that's annoying.

This would have been great to upstream but given that flag it probably can't be folded into main.

By the second iteration of the loop are all functions registered?

Reply:
All the functions are registered when I was testing locally, but I'm not 100% sure they will in prod because we have so many of them.

The code does read like it _should_ register all of them though 🤷 

https://github.com/gadget-inc/fission/pull/21/files#diff-916db856fbad7b791cbe43e341279a46aa846edc27fe09952ca602dc458442a7R384-R405

Reply:
Yeah that looks like it should get everything